packages/table/src/headers/header.tsx

```diff
@@ -239,8 +239,11 @@ export class Header extends React.Component<IInternalHeaderProps, IHeaderState>
         this.state = { hasValidSelection: this.isSelectedRegionsControlledAndNonEmpty(props) };
     }
 
-    public componentWillReceiveProps(nextProps?: IInternalHeaderProps) {
-        this.setState({ hasValidSelection: this.isSelectedRegionsControlledAndNonEmpty(nextProps) });
+    public componentDidUpdate(_: IInternalHeaderProps, prevState: IHeaderState) {
+        const nextHasValidSection = this.isSelectedRegionsControlledAndNonEmpty(this.props);
+        if (prevState.hasValidSelection !== nextHasValidSection) {
```

Comment:
nit: is this check necessary? isn't `this.setState` a no-op if the values are equal?

Reply:
it is. no noop